### DISCUSSION TITLE
dataset api

### DISCUSSION THREAD

**Discussion by sun-timer** [infiniflow/ragflow#2546] [Q&A]
When I use api to create dataset, it returns knowledge-base create successfully:  {'data': None, 'retcode': 100, 'retmsg': "&lt;NotFound '404: Not Found'&gt;"}. Could you help me to check this out ![image](https://github.com/user-attachments/assets/2b4c6ef8-5499-4ed0-b373-2b95d4ce61b5)

**Comment by KevinHuSh** [comment]
The base URL should be /api/v1/

**Reply by sun-timer** [reply]
after i use /api/v1, it returns 405 as following:
[image](https://github.com/user-attachments/assets/d90a33bd-9493-49a4-b3b5-ce92676f8359)

**Comment by tiparamona** [comment]
I tried locally with the following curl command:
 

```
curl    http://www.ragflow.local/api/v1/dataset/knowledge \
     -H "Authorization: Bearer ragflow-token" \
     -H "Content-Type: application/json" \
     -d '{"dataset_name": "My New Dataset"}' \
     -v 
```

And the response I received was:

 
```
* Host www.ragflow.local:80 was resolved.
* IPv6: (none)
* IPv4: 127.0.0.1
*   Trying 127.0.0.1:80...
* Connected to www.ragflow.local (127.0.0.1) port 80
> POST /api/v1/dataset/knowledge HTTP/1.1
> Host: www.ragflow.local
> User-Agent: curl/8.5.0
> Accept: */*
> Authorization: Bearer ragflow-token
> Content-Type: application/json
> Content-Length: 34
> 
< HTTP/1.1 405 Not Allowed
< Server: nginx/1.18.0 (Ubuntu)
< Date: Tue, 24 Sep 2024 15:14:18 GMT
< Content-Type: text/html
< Content-Length: 166
< Connection: keep-alive
< 
<html>
<head><title>405 Not Allowed</title></head>
<body>
<center><h1>405 Not Allowed</h1></center>
<hr><center>nginx/1.18.0 (Ubuntu)</center>
</body>
</html>
* Connection #0 to host www.ragflow.local left intact
```
 

Could you please assist me in creating a dataset through the API?  I have looked through the Git conversations and files, but I haven’t succeeded in creating a dataset through the API. Thank you!

Additionally, I would like to confirm that other API (on v1/api) functionalities are working properly:
Creating a New Conversation: The endpoint for creating a new conversation is functioning correctly and responds as expected.
Uploading a File: The API supports file uploads without any issues.
I appreciate your assistance!

**Reply by Feiue** [reply]
> I tried locally with the following curl command:
> 
> ```
> curl    http://www.ragflow.local/api/v1/dataset/knowledge \
>      -H "Authorization: Bearer ragflow-token" \
>      -H "Content-Type: application/json" \
>      -d '{"dataset_name": "My New Dataset"}' \
>      -v 
> ```
> 
> And the response I received was:
> 
> ```
> * Host www.ragflow.local:80 was resolved.
> * IPv6: (none)
> * IPv4: 127.0.0.1
> *   Trying 127.0.0.1:80...
> * Connected to www.ragflow.local (127.0.0.1) port 80
> > POST /api/v1/dataset/knowledge HTTP/1.1
> > Host: www.ragflow.local
> > User-Agent: curl/8.5.0
> > Accept: */*
> > Authorization: Bearer ragflow-token
> > Content-Type: application/json
> > Content-Length: 34
> > 
> < HTTP/1.1 405 Not Allowed
> < Server: nginx/1.18.0 (Ubuntu)
> < Date: Tue, 24 Sep 2024 15:14:18 GMT
> < Content-Type: text/html
> < Content-Length: 166
> < Connection: keep-alive
> < 
> <html>
> <head><title>405 Not Allowed</title></head>
> <body>
> <center><h1>405 Not Allowed</h1></center>
> <hr><center>nginx/1.18.0 (Ubuntu)</center>
> </body>
> </html>
> * Connection #0 to host www.ragflow.local left intact
> ```
> 
> Could you please assist me in creating a dataset through the API? I have looked through the Git conversations and files, but I haven’t succeeded in creating a dataset through the API. Thank you!
> 
> Additionally, I would like to confirm that other API (on v1/api) functionalities are working properly: Creating a New Conversation: The endpoint for creating a new conversation is functioning correctly and responds as expected. Uploading a File: The API supports file uploads without any issues. I appreciate your assistance!

Please make sure to use `api/v1` as it's the latest API version. The endpoint for creating a knowledge base is `dataset/save`, and for creating a conversation, it's `session/save`. The latest API documentation will be made public as soon as possible.
